robolectric-shadows/shadows-core/src/main/java/org/robolectric/fakes/RoboMenu.java

```diff
@@ -52,7 +51,7 @@ public MenuItem add(int groupId, int itemId, int order, CharSequence title) {
 
   @Override
   public MenuItem add(int groupId, int itemId, int order, int titleRes) {
-    return add(groupId, itemId, order, context.getResources().getString(titleRes));
+    return add(groupId, itemId, order, RuntimeEnvironment.application.getResources().getString(titleRes));
```

Comment:
1. Can you give details on what didn't work? It should work fine so long as the base context is attached :-)
2. There isn't a RoboSubMenuItem (at head least), what class are you referring to?